src/pool/srv_pool.c

```diff
@@ -1540,8 +1540,6 @@ ds_pool_svc_load(struct rdb_tx *tx, uuid_t uuid, rdb_path_t *root, uint32_t *glo
 	*global_version_out = global_version;
 	*map_buf_out = map_buf;
 	*map_version_out = map_version;
-	if (rc != 0)
-		D_FREE(map_buf);
```

Comment:
[Note] This is probably caused by previous merged with master.

Reply:
or set "*map_version_out = NULL" at above L1519 (rc != 0) case, seems in that case the map_buf ptr in stack possibly not init.

Reply:
@liuxuezhao, it is intentional that `ds_pool_svc_load` changes `*map_version_out` **only when returning zero**. Callers shall only expect an output in `*map_version_out` if `ds_pool_svc_load` returns zero.